QMigrate/lut/create_lut.py

```diff
@@ -17,236 +17,246 @@
 from .lut import LUT
 
 
-def read_nlloc(path, stations):
+def read_nlloc(path, stations, phases=["P", "S"], fraction_tt=0.1):
     """
-    Read in a travel-time lookup table that is saved in the NonLinLoc format.
+    Read in a traveltime lookup table that is saved in the NonLinLoc format.
 
     Parameters
     ----------
     path : str
         Path to directory containing .buf and .hdr files.
-    stations : pandas DataFrame
+    stations : `pandas.DataFrame`
         DataFrame containing station information (lat/lon/elev).
+    phases : list of str, optional
+        List of seismic phases for which to calculate traveltimes.
+    fraction_tt : float, optional
+        An estimate of the uncertainty in the velocity model as a function of
+        a fraction of the traveltime.
 
     Returns
     -------
-    lut : QuakeMigrate lookup table object
-        Lookup table populated with travel times from the NonLinLoc files.
+    lut : `QMigrate.lut.LUT` object
+        Lookup table populated with traveltimes from the NonLinLoc files.
 
     """
 
     path = pathlib.Path(path)
 
-    for i, station in stations.iterrows():
-        name = station["Name"]
-        print(f"Loading P- and S- travel-time lookup tables for {name}")
-
-        p_file = path / f"layer.P.{name}.time"
-        s_file = path / f"layer.S.{name}.time"
-
-        if i == 0:
-            gridspec, transform, pttimes = _read_nlloc(p_file)
-            _, _, sttimes = _read_nlloc(s_file)
-            cell_count = np.array(gridspec[0])
-            grid_origin = np.array(gridspec[1]) * 1000
-            cell_size = np.array(gridspec[2]) * 1000
-
-            gproj, cproj = transform
-            if gproj is None:
-                raise NotImplementedError
+    print("Loading NonLinLoc traveltime lookup tables for...")
+    for i, phase in enumerate(phases):
+        print(f"\t...phase: {phase}...")
+        for j, station in enumerate(stations["Name"].values):
+            print(f"\t\t...station: {station}")
+            file = path / f"layer.{phase}.{station}.time"
+
+            if i == 0 and j == 0:
+                gridspec, transform, traveltimes = _read_nlloc(file)
+                cell_count = np.array(gridspec[0])
+                grid_origin = np.array(gridspec[1])
+                cell_size = np.array(gridspec[2])
+
+                gproj, cproj = transform
+                if gproj is None:
+                    raise NotImplementedError
```

Comment:
We do support reading the projection, but only `SIMPLE`, `LAMBERT`, `TRANS_MERC` - but yeah, perhaps an unsupported projection type in the NonLinLoc header file?

Reply:
Yep, sounds good. If the error is isolated to that cause, then seems worth including in the `NotImplementedError`.

Maybe: 
```suggestion
                    raise NotImplementedError(f"The projection type {blah} is not currently supported. "
                                              f"The available options are 'SIMPLE', 'LAMBERT' and 'TRANS_MERC'. "
                                              "Please contact the QuakeMigrate developers.")
```

This will require passing the projection string up from `_read_nlloc()`